plugins/inputs/cisco_telemetry_gnmi/cisco_telemetry_gnmi.go

```diff
@@ -101,11 +101,13 @@ func (c *CiscoTelemetryGNMI) Start(acc telegraf.Accumulator) error {
 	// Invert explicit alias list and prefill subscription names
 	c.aliases = make(map[string]string, len(c.Subscriptions)+len(c.Aliases))
 	for _, subscription := range c.Subscriptions {
-		path := subscription.Path
-		if len(subscription.Origin) > 0 {
-			path = subscription.Origin + ":" + path
+		var path string
+		// Build the subscription path without keys
+		if parsed, _ := parsePath(subscription.Origin, subscription.Path, ""); parsed != nil {
```

Comment:
If an error occurs here, it seems like we return it because that means the path is invalid in the configuration and an invalid config value should be fatal.